### DISCUSSION TITLE
group_replication_transaction_size_limit error when attempting a basic mydumper/myloader

### DISCUSSION THREAD

**Discussion by matthew-lenz-otr** [mydumper/mydumper#1692] [Q&A]
Any recommendations on how to avoid something like this in the future?  no preference if it's a mydumper or myloader flag I need to provide.

**Comment by davidducos** [comment]
Hi @matthew-lenz-otr, 
What was the issue? was it on mydumper? or in myloader? 
You can configure session and global variables for mydumper and myloader on the configuration file. Documentation is on its way...

**Comment by matthew-lenz-otr** [comment]
It happened with myloader.  I don't think mysql group replication setups are all that rare anymore.  This exceeded the default maximum default transaction size.   If this is can be controlled via myloader would it make sense for myloader to check that variable and automatically adjust its own settings?  I assume that the transaction size can be controlled in myloader?

**Comment by davidducos** [comment]
Hi @matthew-lenz-otr, 
We don't have a particular variable to control the transaction size on myloader, but we have --queries-per-transaction that can reduce the transaction size, and if you set to 0, it will disable transactions and you have also --rows which will split the inserts per amount of rows. A combination of both variable might help you to workaround the group_replication_transaction_size_limit error.

Another thing to consider is that the default statement size on mydumper is 1MB and the default of --queries-per-transaction on myloader is 1000, so we might end up with transactions of 1GB which is larger than the 150MB default size on group_replication_transaction_size_limit. We could add on mydumper a value in the metadata file with the largest statement length written so myloader could estimate a safe value for --queries-per-transaction by default (replacing the static value of 1000) taking into account the group_replication_transaction_size_limit. 
This sound complex but actually, it might be simpler to implement as mydumper is already sharing info so myloader can perform better imports.

**Comment by matthew-lenz-otr** [comment]
@davidducos whatever you think is best.  I know you've been striving to make mydumper/myloader "just work" as much as possible and thought others might run into this situation as mysql group replication setups become more popular.

**Reply by davidducos** [reply]
I created https://github.com/mydumper/mydumper/issues/1694 to follow this request
